src/api/goods.ts

```diff
@@ -3,17 +3,27 @@ import { Good } from '../types/Good';
 // eslint-disable-next-line
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
-export function getAll(): Promise<Good[]> {
-  return fetch(API_URL)
-    .then(response => response.json());
+export async function getAll(): Promise<Good[]> {
+  const allGoods = await fetch(API_URL);
+
+  if (!allGoods.ok) {
+    throw new Error('Cannot fetch goods');
+  }
+
+  return allGoods.json();
 }
 
-export const get5First = () => {
-  return getAll()
-    .then(goods => goods); // sort and get the first 5
+export const get5First = async () => {
+  const allGoods = await getAll();
+
+  return allGoods
+    .sort((firstGood, secondGood) => (
+      firstGood.name.localeCompare(secondGood.name)))
+    .slice(0, 5);
 };
 
-export const getRedGoods = () => {
-  return getAll()
-    .then(goods => goods); // get only red
+export const getRedGoods = async () => {
+  const allGoods = await getAll();
+
+  return allGoods.filter((good) => good.color === 'red');
```

Comment:
```suggestion
  return allGoods.filter(({ color }) => color === 'red');
```

Reply:
fixed all comment without demo link? and I have not idea why it dosent work? wrote to slack qna